place-work/v1alpha1/types.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	work "open-cluster-management.io/api/work/v1"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// PlaceManifestWorkSpec defines the desired state of PlaceManifestWork
+type PlaceManifestWorkSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// ManifestWorkSpec is the ManifestWorkSpec that will be used to generate a per-cluster ManifestWork
+	ManifestWorkSpec work.ManifestWorkSpec `json:"manifestWorkSpec"`
+
+	// PacementRef is the name of the Placement resource, from which a PlacementDecision will be found and used
+	// to distribute the ManifestWork
+	PlacementRef *corev1.LocalObjectReference `json:"placementRef,omitempty"`
+}
+
+// PlaceManifestWorkStatus defines the observed state of PlaceManifestWork
+type PlaceManifestWorkStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Conditions contains the different condition statuses for distrbution of ManifestWork resources
+	// Valid condition types are:
+	// 1. AppliedManifestWorks represents ManifestWorks have been distributed as per placement All, Partial, None, Problem
+	// 2. PlacementRefValid
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// ManifestWorkDelivery tracks each ManifestWork that was created,
+	// including that it was applied and its overall status
+	PlacedManifestWork []PlacedManifestWork `json:"placedManifestWork,omitempty"`
```

Comment:
This could become to much data and hard to scale. why not just giving the status per cluster ex; applied/processing/available/degraded and for more info used can check the manifestWork itself. What do u think ?

Reply:
I have to go back and check the synthetic scale test on the Controller when I put that pull request in, but i was able to handle distribution to 2000 clusters at the time.

Reply:
Pretty much its 2000 X (ObjectRef + bool + timestamp + string)
It's a finite sized object, if we get the cluster count to high though it might become a problem.  When I revisit the synthetic scale test, I'll try it with 3000-4000 clusters to make sure and also work out the calculation.

Reply:
I think we might start without this field. IT could be too many items and introduce many update calls

Reply:
So i've been thinking about options to trigger the updates via a backoff algorithm, which I still think is problematic, or by triggering the Place-Work-controller, but making sure not to trigger it if the last time it ran is Less Than.

1. I'm pretty sure the size was not a factor in my synthetic scale test as its a fixed size.
2. Initial controller just reconciles on a time loop, like ManifestWork.

Reply:
but who will consume this field? It seems to me that having summary field is sufficient, and since we can build a linkage between placementWork and manifestwork, user can get more details from manifestwork. Similar as deployment/pod.

Reply:
may be just return the `manifestWork.Status.ResourceStatus.StatusFeedback` If the user defines the feedbackRules in the `ManifestWork.spec` , what do u think ? otherwise I'm more with Jian its too much info that will cause issues than help

Reply:
The StatusFeedback from my perspective is a bigger risk then These details, but I'll remove them for now.